qabelbox/src/androidTest/java/de/qabel/qabelbox/persistence/RepositoryFactoryTest.java

```diff
@@ -0,0 +1,61 @@
+package de.qabel.qabelbox.persistence;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+import android.test.RenamingDelegatingContext;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import de.qabel.desktop.repository.sqlite.AndroidClientDatabase;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+@RunWith(AndroidJUnit4.class)
+public class RepositoryFactoryTest{
+
+    Context mMockContext;
+    private RepositoryFactory repositoryFactory;
+
+    @Before
+    public void setUp() {
+        mMockContext = new RenamingDelegatingContext(
+                InstrumentationRegistry.getInstrumentation().getTargetContext(), "factorytest_");
+        repositoryFactory = new RepositoryFactory(mMockContext);
+        repositoryFactory.close();
+        repositoryFactory.deleteDatabase();
+    }
+
+    @After
+    public void tearDown() {
+        repositoryFactory.close();
+        repositoryFactory.deleteDatabase();
+    }
+
+    @Test
+    public void testCanDeleteDatabase() throws Exception {
+        repositoryFactory.getAndroidClientDatabase().setVersion(1);
```

Comment:
is this test still of value? You are actually just testing nothing. the setVersion is never asserted with a getVersion, the `getAndroidClientDatabase()` method is already called dozens of times by other tests and the `deleteDatabase()` is only for test purposes according to it's comment and the only validation here is that it doesn't throw an exception so it could be empty to pass this test...